### ISSUE TITLE
Duplicate Symbols when Podfile contain both root pods and `target`-specific pods

### ISSUE THREAD

**Issue by AliSoftware** [CocoaPods/CocoaPods#1729]
> CocoaPods version: `0.29.0` (but not sure if already present in earlier versions)
> NB: Maybe related to #1674 and/or #1688?

## Context

Let say we have the following `Podfile`:

```
pod 'Common'

target "MyApp" do
  pod 'Foo'
end

target "MyAppTests" do
  pod 'Bar'
end
```

:arrow_right: This leads to duplicate symbols for the files from the `"Common"` pod:
![actualresult](https://f.cloud.github.com/assets/216089/1845987/66ac0d56-75bb-11e3-8eae-11f32566a74a.png)

## Expected Result

We expect that CocoaPods would then create the following targets in `Pods.xcodeproj`:

* `Pods`: "aggregate" target that simply links with `libPods-Common.a`
* `Pods-Common`: target that builds the lib associated with the `Common` pod
* `Pods-MyApp`: "aggregate" target that simply links with `libPods-MyApp-Foo.a`
* `Pods-MyApp-Foo`: target that builds the lib associated with the `Foo` pod
* `Pods-MyAppTests`: "aggregate" target that simply links with `libPods-MyAppTests-Bar.a`
* `Pods-MyAppTests-Bar`: target that build the lib associated with the `Bar` pod

And then the `MyApp.xcodeproj` would have:
* its "MyApp" target link against `libPods.a` AND `libPods-MyApp.a`
* its "MyAppTests" target link against `libPods.a` AND `libPods-MyAppTests.a``
.

## Actual Result

In addition to the previously listed ones, the following targets are also generated in `Pods.xcodeproj`:

* `Pods-MyApp-Common`
* `Pods-MyAppTests-Common`

_Both compile the files of `Common`pod and are duplicates of what the `Pod-Common` target do_

The `Pods-MyApp` "aggregate" lib target then links against `libPods-MyApp-Common.a` in addition to `libPods-MyApp-Foo.a`, as well as the `Pods-MyAppTests` "aggregate" lib target links against `libPods-MyAppTest-Common.a` in addition to `libPods-MyAppTests-Bar.a`

**[This leads to duplicate symbols](https://f.cloud.github.com/assets/216089/1845987/66ac0d56-75bb-11e3-8eae-11f32566a74a.png), as the files in the `Common` pod/library are linked twice, once with `libPods-Common.a` and once with `libPods-MyApp[Tests]-Common.a`** whereas they should appear in only one aggregate.

---

The solution is:

* [Keep `libsPods.a` and `libPods-Common.a`](https://f.cloud.github.com/assets/216089/1846001/7b170f56-75bc-11e3-9759-3b11fdc9c787.png) — and link both "MyApp" and "MyAppTests" against it in addition to linking them against their `libPods-MyApp[Tests].a` specific aggregate libs — but stop generating `libPods-MyApp[Tests]-Common.a`
* Or [get rid of the `libPods.a` and `libPods-Common.a`](https://f.cloud.github.com/assets/216089/1845998/289bf796-75bc-11e3-81bf-216b0b8794a1.png) targets/libraries and stop linking `MyApp` and `MyAppTests` against it — but keep `libPods-MyApp[Tests]-Common.a`

> The first solution seems much more logical to me as it would reflect the configuration described in the `Podfile`, whereas the last solution better corresponds to the case where the user would have repeated `pod "Common"` in each `target XXX do` block)

**Comment by manicaesar** [comment]
@alexburkhay In our project we found one another workaround. If your primary target is a library, which is then linked against by some App, then the solution is to ... **make the library a pod itself** ;) And link the App to the pod (preferably in a form of development pod).

Example:
```
Primary target: MyLibrary
Another target: MyApp (uses MyLibrary)
```

Podfile (implicit target is assumed to be MyLibrary)
```
pod 'X'
pod 'Y'

target :MyApp, :exclusive => true do
    pod 'MyLibrary', :path => '.' #or wherever podspec for your MyLibrary is
    pod 'Z'
end
```
NOTE: Please remember to **not** add libMyLibrary.a to the list of linked binaries of MyApp. Pods will do it for you.

**Comment by cbowns** [comment]
@AliSoftware Hmmm, clever solution. What's weird is, I get an error where my main target tries to link `-lPods` anyhow. I've dug through my `.xcconfig`s and project linker flags and simply can't find where that flag comes from. Did you experience this, or no?

**Comment by segiddins** [comment]
This should be fixed by #3550, which is in 0.38.
